src/NodaTime.Web/Startup.cs

```diff
@@ -129,5 +136,40 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerF
             // Force the set of releases to be first loaded on startup.
             app.ApplicationServices.GetRequiredService<IReleaseRepository>().GetReleases();
         }
+
+        /// Sets the Cache-Control header for static content, conditionally allowing the browser to use the content
+        /// without revalidation.
+        private void SetCacheControlHeaderForStaticContent(IHostingEnvironment env, HttpContext context)
+        {
+            var headers = new ResponseHeaders(context.Response.Headers);
+
+            if (headers.ContentType.IsSubsetOf(TextHtml))
+            {
+                // Don't set Cache-Control for HTML files (e.g. /tzdb/). The browser can figure out when to revalidate.
+                // (Which it can do easily, since we send an ETag with all static content responses).
+                return;
+            }
+
+            // Otherwise if the request was made with a file version query parameter (?v=..., as sent by
```

Comment:
True, but I think the separate version is a bit more readable here (and the minor duplication doesn't bother me too much).